### PR TITLE
adding keyPurchase to walletService integration tests

### PR DESCRIPTION
**Pull request by julien51**
# Description

This is the next step on the journey to make unlockjs very robust!

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-js/src/__tests__/integration/walletServiceIntegration.test.js [hunk 1]
```diff
@@ -138,7 +139,111 @@ describe('Wallet Service Integration', () => {
         })
       })
 
-      describe('purchaseKey', () => {})
+      describe('purchaseKey', () => {
+        let tokenId
+        let key
+        let owner, sender
```
Comment (cnasc) [inline unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: Can we make these names a bit more specific? I wasn't sure what these roles were until I'd read the tests a few times. Maybe something like `keyPurchaser` and `keyRecipient`?

File: unlock-js/src/__tests__/integration/walletServiceIntegration.test.js [hunk 2]
```diff
@@ -138,7 +139,111 @@ describe('Wallet Service Integration', () => {
...
+            )
+          } else {
+            // For ERC20 the balance should be exact
+            expect(parseFloat(newBalance)).toBe(
+              parseFloat(userBalanceBefore) - parseFloat(lock.keyPrice)
+            )
+          }
+        })
+
+        it('should have assigned the key to the right user', async () => {
```
Comment (cnasc) [inline unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: I think you have the descriptions for this test and the following one swapped

File: unlock-js/src/__tests__/integration/walletServiceIntegration.test.js (hunk 1 above)
Comment (julien51) [reply unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: yes!

File: unlock-js/src/__tests__/integration/walletServiceIntegration.test.js (hunk 2 above)
Comment (julien51) [reply unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: oups.

Comment (julien51) [reply unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: fixed!

**Comment by julien51** [comment]
I still have a problem with ERC20 locks apparently.

File: unlock-js/src/__tests__/integration/walletServiceIntegration.test.js [hunk 3]
```diff
@@ -122,7 +123,7 @@ describe('Wallet Service Integration', () => {
 
       describe('updateKeyPrice', () => {
         let oldKeyPrice, newPrice
-        beforeEach(async () => {
+        beforeAll(async () => {
```
Comment (julien51) [inline unlock-js/src/__tests__/integration/walletServiceIntegration.test.js]: This is a cosmtic fix since there is only a single test.

**Comment by julien51** [comment]
ok, PTAL @cnasc ?

**Comment by julien51** [comment]
Hum, it looks like the purchaser account is not getting their ERC20.

**Pull request merged**